flac.md

```diff
@@ -544,61 +544,28 @@ Data      | Description
-`u(4(+5))` | `EXP-GOLOMB PARTITION ENCODING PARAMETER` (see [section on EXP-GOLOMB PARTITION ENCODING PARAMETER](#exp-golomb-partition-encoding-parameter))
-`u(?)`     | `ENCODED RESIDUAL` (see [section on ENCODED RESIDUAL](#encoded-residual))
+Each partition contains a certain amount of residual samples. The number of residual samples in the first partition is equal to (blocksize >> partition order) - predictor order, i.e. the blocksize divided by the number of partitions minus the predictor order. In all other partitions the number of residual samples is equal to (blocksize >> partition order).
 
-#### EXP GOLOMB PARTITION ENCODING PARAMETER
-Value           | Description
----------------:|:-----------
-0b0000 - 0b1110 | Exp-golomb parameter.
-0b1111          | Escape code, meaning the partition is in unencoded binary form using n bits per sample; n follows as a 5-bit number.
-
-### RESIDUAL_CODING_METHOD_PARTITIONED_EXP_GOLOMB2
-Data               | Description
-:------------------|:-----------
-`u(4)`             | Partition order.
-`EXP-GOLOMB2_PARTITION`+ | There will be 2\^order partitions.
+The partition order MUST be so that the blocksize is evenly divisible by the number of partitions. This means for example that for all odd blocksizes, only partition order 0 is allowed.  The partition order also MUST be so that the (blocksize >> partition order) is larger than the predictor order. This means for example that with a blocksize of 4096 and a predictor order of 4, partition order cannot be larger than 9.
 
-#### EXP_GOLOMB2_PARTITION
-Data       | Description
-:----------|:-----------
-`u(5(+5))` | `EXP-GOLOMB2 PARTITION ENCODING PARAMETER` (see [section on EXP-GOLOMB2 PARTITION ENCODING PARAMETER](#expgolomb2-partition-encoding-parameter))
-`u(?)`     | `ENCODED RESIDUAL` (see [section on ENCODED RESIDUAL](#encoded-residual))
+In case the coded residual of a subframe is one with a 4-bit Rice parameter (see table at the start of this section), the first 4 bits of each partition are either a rice parameter or an escape code. These 4 bits indicate an escape code if they are 0b1111, otherwise they contain the rice parameter as an unsigned number. In case the coded residual of the current subframe is one with a 5-bit Rice parameter, the first 5 bits indicate an escape code if they are 0b11111, otherwise they contain the rice parameter as an unsigned number as well.
 
-#### EXP-GOLOMB2 PARTITION ENCODING PARAMETER
-Value             | Description
------------------:|:-----------
-0b00000 - 0b11110 | Exp-golomb parameter.
-0b11111           | Escape code, meaning the partition is in unencoded binary form using n bits per sample; n follows as a 5-bit number.
+In case an escape code was used, the partition does not contain a variable-length rice coded residual, but a fixed-length unencoded residual. Directly following the escape code are 5 bits containing the number of bits with which each residual sample is stored, as an unsigned number.
```

Comment:
```suggestion
In case an escape code was used, the partition does not contain a variable-length rice coded residual, but a fixed-length unencoded residual, with each residual sample signed two's complement. Directly following the escape code are 5 bits containing the number of bits with which each residual sample is stored, as an unsigned number.
```